examples/anthropic/anthropic_example.ipynb

```diff
@@ -0,0 +1,175 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we'll grab our API keys. You can use dotenv like below or however else you like to load environment variables"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "load_dotenv()\n",
+    "ANTHROPIC_API_KEY = os.getenv(\"ANTHROPIC_API_KEY\") or \"<your_anthropic_key>\"\n",
+    "AGENTOPS_API_KEY = os.getenv(\"AGENTOPS_API_KEY\") or \"<your_agentops_key>\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "agentops.init(AGENTOPS_API_KEY, default_tags=[\"anthropic-example\"])\n",
+    "client = anthropic.Anthropic()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "stream = client.messages.create(\n",
+    "    max_tokens=1024,\n",
+    "    model=\"claude-3-opus-20240229\",\n",
+    "    messages=[\n",
+    "        {\n",
+    "            \"role\": \"user\",\n",
+    "            \"content\": \"What is 2+2?\",\n",
+    "        }\n",
+    "    ],\n",
+    "    stream=True,\n",
+    ")\n",
+    "\n",
+    "response = \"\"\n",
+    "for event in stream:\n",
+    "    if event.type == \"content_block_delta\":\n",
+    "        response += event.delta.text\n",
+    "    elif event.type == \"message_stop\":\n",
+    "        print(\"\\n\")\n",
+    "        print(response)\n",
+    "        print(\"\\n\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "stream = client.messages.create(\n",
+    "    max_tokens=1024,\n",
+    "    model=\"claude-3-opus-20240229\",\n",
+    "    tools=[\n",
+    "        {\n",
+    "            \"name\": \"web_search\",\n",
+    "            \"description\": \"Search the web for information\",\n",
+    "            \"input_schema\": {\n",
+    "                \"type\": \"object\",\n",
+    "                \"properties\": {\n",
+    "                    \"query\": {\"type\": \"string\", \"description\": \"The search query\"}\n",
+    "                },\n",
+    "                \"required\": [\"query\"],\n",
+    "            },\n",
+    "        }\n",
+    "    ],\n",
+    "    messages=[\n",
+    "        {\n",
+    "            \"role\": \"user\",\n",
+    "            \"content\": \"Tell me everything you can about AgentOps\",\n",
+    "        }\n",
+    "    ],\n",
+    "    stream=True,\n",
+    ")\n",
+    "\n",
+    "response = \"\"\n",
+    "for event in stream:\n",
+    "    if event.type == \"content_block_delta\":\n",
+    "        if event.delta.type == \"text\":\n",
+    "            response += event.delta.text\n",
+    "    elif event.type == \"message_stop\":\n",
+    "        print(\"\\n\")\n",
+    "        print(response)\n",
+    "        print(\"\\n\")"
```

Comment:
I re-ran the code and found out the tool calling isn't doing what it's expected, in this case, performing a web search). I have 2 solutions in mind for a fix -
* Precisely describe the web search in the tool
* Use the [anthropic-tools](https://github.com/anthropics/anthropic-tools) package which is in alpha stage

I will be testing both of these and push a fix soon.

Reply:
I tested with both of these and so far no luck since there is no innate web search support for the models. However, another solution would be to use the anthropic-tools package with a custom Python class that uses a package to perform web searches, getting us the results we want. Or we could skip such questions and ask something else in the prompt.

Please let me know what are your thoughts on this.